src/parser/mage/frost/modules/features/WaterElemental.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import Icon from 'common/Icon';
+import { formatPercentage, formatNumber, formatThousands } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import Analyzer from 'parser/core/Analyzer';
+import DamageValue from 'parser/shared/modules/DamageValue';
+
+class Waterbolt extends Analyzer {
+  
+  constructor(...args) {
+    super(...args);
+    this.active = !this.selectedCombatant.hasTalent(SPELLS.LONELY_WINTER_TALENT.id);
+  }
+
+  get suggestionThresholds() {
+    return {
+      actual: this.activeTimePercentage,
+      isLessThan: {
+        minor: 0.95,
+        average: 0.75,
+        major: 0.50,
+      },
+      style: 'percentage',
+    };
+  }
+  
+  get downtimePercentage() {
+    return 1 - this.activeTimePercentage;
+  }
+
+  get activeTimePercentage() {
+    return this.activeTime / this.owner.fightDuration;
+  }
+
+  _waterboltsCancelled = 0;
+  _waterboltsCastStarts = 0;
+  _waterboltHits = 0;
+  wasCastStarted;
+  activeTime = 0;
+  beginCastSpell = 0;
+  _timestampLastFinish = 0;
+  _timestampLastCast = 0;
+
+  _byPet = {};
+  byPet(petId) {
+    if (!this._byPet[petId]) {
+      return new DamageValue();
+    }
+    return this._byPet[petId];
+  }
+
+  get totalByPets() {
+    return Object.keys(this._byPet)
+      .map(petId => this._byPet[petId])
+      .reduce((total, damageValue) => total.add(damageValue.regular, damageValue.absorbed, damageValue.blocked, damageValue.overkill), new DamageValue());
+  }
+
+  on_byPlayerPet_damage(event) {
+    if (!event.targetIsFriendly) {
+      const petId = event.sourceID;
+      this._byPet[petId] = this.byPet(petId).add(event.amount, event.absorbed, event.blocked, event.overkill);
+      if(event.ability.guid === SPELLS.WATERBOLT.id) {
+        this._waterboltHits += 1;
+      }
+    }
+  }
+
+  on_byPlayerPet_cast(event) {
+    if (event.ability.guid === SPELLS.WATERBOLT.id) {
```

Comment:
We prefer to exit-early in these event listeners rather than use multlevel if statements:
if(guid != WATERBOLT) {
  return;
}